### PR TITLE
minor comments

### PR DESCRIPTION
**Pull request by omnifient**
gm, overall looks good, just some minor comments

also missing license (use unlicensed or whatever polygon prefers?)

### CODE REVIEW

File: src/MiniExec.sol [hunk 1]
```diff
@@ -31,62 +35,37 @@ contract MiniExecImplementation is IZkEVMBridgeMessageReceiver {
...
 contract MiniExecFactory {
-    struct Metadata {
-        address owner;
-        uint32 networkId;
-    }
-
-    mapping(address => Metadata) public metadata;
-    mapping(address => address) public implementations;
-
     MiniExecImplementation public immutable miniExecImplementation;
```
Comment (omnifient) [inline src/MiniExec.sol]: hmm just we also have a way to change this implementation?

File: test/MiniExec.t.sol [hunk 2]
```diff
@@ -10,7 +12,21 @@ contract AnotherImplementation {
...
 
-contract MiniExecTest is Test, MiniExecScript {
+contract MockContract {
+    uint256 public a;
+
+    function increment() external {
+        a += 1;
+    }
+
+    function decremenet() external {
```
Comment (omnifient) [inline test/MiniExec.t.sol]: typo: *decrement

File: test/MiniExec.t.sol [hunk 3]
```diff
@@ -23,37 +37,55 @@ contract MiniExecTest is Test, MiniExecScript {
...
-
-        constructor(address a) {
-            referenceAddr = a;
-        }
-
-        function someFunction(address a, uint256 u) {
-            assert(a == referenceAddr);
-            referenceValue = u;
-        }
+    function testCallsCorrectly() external {
```
Comment (omnifient) [inline test/MiniExec.t.sol:45]: the goal with `someFunction` was that you'd test something that passes values and changes the storage ;)

File: src/MiniExec.sol [hunk 4]
```diff
@@ -35,71 +35,37 @@ contract MiniExecImplementation is IZkEVMBridgeMessageReceiver {
...
     MiniExecFactory immutable creator;
 
-    // @OF: spec says "Constructor takes in tuple of address and chain ID"
-    constructor() {
-        creator = MiniExecFactory(msg.sender);
-    }
-
-    function updateImplementation(address _newImplementation) external onlySelf {
-        creator.setImplementation(_newImplementation);
+    constructor(address owner, uint32 networkId, address impl) {
```
Comment (omnifient) [inline src/MiniExec.sol]: note: while this works, the common pattern for setting the values in the implementation is to pass the values through an initializer

File: test/MiniExec.t.sol (hunk 3 above)
Comment (0xdapper) [reply test/MiniExec.t.sol:45]: can do that, but this tests that calldata is passed correctly, no difference between calldata being just a selector or selector and encoded params

File: src/MiniExec.sol (hunk 4 above)
Comment (0xdapper) [reply src/MiniExec.sol]: yeah but comment/spec was specifically for a constructor?

Comment (0xdapper) [reply src/MiniExec.sol]: moved to initializer

File: test/MiniExec.t.sol (hunk 3 above)
Comment (0xdapper) [reply test/MiniExec.t.sol:45]: added a `setB` function

File: src/MiniExec.sol (hunk 1 above)
Comment (0xdapper) [reply src/MiniExec.sol]: added

**Comment by omnifient** [comment]
it's ok to merge

**Pull request merged**